content/blog/python-for-pac/index.md

```diff
@@ -0,0 +1,205 @@
+*mandatory*, which prevents a resource deployment if it violates the policy.
+
+Pulumi supports two types of policies:
+
+- **ResourceValidationPolicy** validates individual resources in a stack and is run during preview.
+- **StackValidationPolicy** validates the stack as a whole and runs when the stack is deployed because some resources may need to exists to validate inputs or outputs.
+
+ResourceValidationPolicy and StackValidationPolicy both run during previews and updates.
+
+ResourceValidationPolicy validates inputs of individual resources before the resource is created or modified. StackValidationPolicy validates outputs of all resources in the stack after all resources have been created or modified. During previews, resources aren't created or modified, so the ResourceValidationPolicy and StackValidationPolicy validations see a preview of what's going to happen (as best we can determine)
+
+With stack validation policies, the validation happens after resources have been created or modified. Modifications to resources can't be prevented, but using mandatory enforcement results in a deployment failing (despite the modified resources). If you run Pulumi preview before an update and use a mandatory StackValidationPolicy, you can catch the problems before a real deployment occurs.
+
+The difference is described in this table:
+
+|                                | Resource Validation                   | Stack Validation                                                                     |
+|--------------------------------|---------------------------------------|--------------------------------------------------------------------------------------|
+| What does it check?            | Individual resources                  | All resources in the stack                                                           |
+| When is the check performed?   | Before resources are created/modified | After all stack resources have been created/modified                                 |
+| What information is available? | Resource _input_ properties           | Resource _output_ properties (Note: inputs are propagated to outputs during preview) |
+
+Now that we have the basics of policy as code let's take a look at example policies written in Python.
+
+## Examples
+
+Policies are validation functions that validate resources in a Pulumi stack. In the examples below, we define a function that takes `ResourceValidationArgs` and `ReportViolation` as arguments. The function checks to see what type of resource and its arguments, these are used to determine if the resource violates the policy. If the resource is not compliant, the function reports the violation, and because the enforcement level is *mandatory*, the resource will not be deployed.
+
+In the case of this Kubernetes policy example, the policy does not allow deploying a load balancer, which would open the kubernetes cluster to the public Internet.
```

Comment:
Is this sentence needed now that k8s is in the chooser? None of the other clouds have similar sentences.

```suggestion
```